tests/test_api_callable.py

```diff
@@ -141,6 +141,18 @@ def test_call_kwargs(self):
         self.assertEqual(my_callable(None, CallOptions(key='updated')),
                          'updated')
 
+    def test_call_merge_options_metadata(self):
+        this_kwargs = {'key': 'value', 'metadata':
+                       [('key1', 'val1'), ('key2', 'val2')]}
+        settings = _CallSettings(kwargs=this_kwargs)
+        my_callable = api_callable.create_api_call(
+            lambda _req, _timeout, **kwargs: kwargs, settings)
+        self.assertEqual(my_callable(None), this_kwargs)
+        this_kwargs['metadata'] = [('key1', 'updated_val1'), ('key2', 'val2')]
```

Comment:
`this_kwargs` is mixing the concepts of input values and expected values. This test would be more readable if we kept those two concepts separate.

Also, I'm a fan of treating variables as immutable whenever it's easy to do so.